File: java-advent-of-code/src/test/java/advent/twenty_eighteen/Day01ChronalCalibrationTest.java
package advent.twenty_eighteen;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class Day01ChronalCalibrationTest {
    @Test
    public void checkReal() {
        Day01ChronalCalibration cc = new Day01ChronalCalibration("/twenty_eighteen/Day01-ChronalCalibration-input.txt");

        assertEquals(430, cc.getPart1Answer().intValue());
        assertEquals(462, cc.getPart2Answer().intValue());
    }

    @Test
    public void checkExample1() {
        Day01ChronalCalibration cc = new Day01ChronalCalibration("/twenty_eighteen/Day01-ChronalCalibration-example#1.txt");

        assertEquals(3, cc.getPart1Answer().intValue());
        assertEquals(2, cc.getPart2Answer().intValue());
    }

    @Test
    public void checkExample2() {
        Day01ChronalCalibration cc = new Day01ChronalCalibration("/twenty_eighteen/Day01-ChronalCalibration-example#2.txt");

        assertEquals(0, cc.getPart1Answer().intValue());
        assertEquals(0, cc.getPart2Answer().intValue());
    }

    @Test
    public void checkExample3() {
        Day01ChronalCalibration cc = new Day01ChronalCalibration("/twenty_eighteen/Day01-ChronalCalibration-example#3.txt");

        assertEquals(4, cc.getPart1Answer().intValue());
        assertEquals(10, cc.getPart2Answer().intValue());
    }

    @Test
    public void checkExample4() {
        Day01ChronalCalibration cc = new Day01ChronalCalibration("/twenty_eighteen/Day01-ChronalCalibration-example#4.txt");

        assertEquals(4, cc.getPart1Answer().intValue());
        assertEquals(5, cc.getPart2Answer().intValue());
    }

    @Test
    public void checkExample5() {
        Day01ChronalCalibration cc = new Day01ChronalCalibration("/twenty_eighteen/Day01-ChronalCalibration-example#5.txt");

        assertEquals(1, cc.getPart1Answer().intValue());
        assertEquals(14, cc.getPart2Answer().intValue());
    }
}

File: kotlin-advent-of-code/src/test/kotlin/advent/twenty_eighteen/Day01ChronalCalibrationTest.kt
package advent.twenty_eighteen

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class Day01ChronalCalibrationTest {
        @Test
        fun checkReal() {
            val cc = Day01ChronalCalibration("/twenty_eighteen/Day01-ChronalCalibration-input.txt")

            assertEquals(430, cc.part1Answer)
            assertEquals(462, cc.part2Answer)
        }

        @Test
        fun checkExample1() {
            val cc = Day01ChronalCalibration("/twenty_eighteen/Day01-ChronalCalibration-example#1.txt")

            assertEquals(3, cc.part1Answer)
            assertEquals(2, cc.part2Answer)
        }

        @Test
        fun checkExample2() {
            val cc = Day01ChronalCalibration("/twenty_eighteen/Day01-ChronalCalibration-example#2.txt")

            assertEquals(0, cc.part1Answer)
            assertEquals(0, cc.part2Answer)
        }

        @Test
        fun checkExample3() {
            val cc = Day01ChronalCalibration("/twenty_eighteen/Day01-ChronalCalibration-example#3.txt")

            assertEquals(4, cc.part1Answer)
            assertEquals(10, cc.part2Answer)
        }

        @Test
        fun checkExample4() {
            val cc = Day01ChronalCalibration("/twenty_eighteen/Day01-ChronalCalibration-example#4.txt")

            assertEquals(4, cc.part1Answer)
            assertEquals(5, cc.part2Answer)
        }

        @Test
        fun checkExample5() {
            val cc = Day01ChronalCalibration("/twenty_eighteen/Day01-ChronalCalibration-example#5.txt")

            assertEquals(1, cc.part1Answer)
            assertEquals(14, cc.part2Answer)
        }
}